backend/api/serializers.py

```diff
@@ -135,6 +135,94 @@ class Meta:
         fields = ('id', 'owner', 'vote')
 
 
+class CorporateUserSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = CorporateUserProfile
+        fields = ('description', 'url', 'location')
+
+
+class UserInterestSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = UserInterest
+        fields = ('name',)
+
+
+class UserCreateSerializer(serializers.ModelSerializer):
+    corporate_profile = CorporateUserSerializer()
+
+    class Meta:
+        model = User
+        fields = ('username', 'email', 'password', 'first_name', 'last_name',
+                  'birth_date', 'is_corporate_user', 'corporate_profile')
+        extra_kwargs = {'password': {'write_only': True}}
+    
+    def create(self, validated_data):
+        print(validated_data)
+        is_corporate_user = validated_data.pop('is_corporate_user')
+        corporate_profile = validated_data.pop('corporate_profile')
+
+        username = validated_data.pop('username')
+        email = validated_data.pop('email')
+        password = validated_data.pop('password')
+        user = User.objects.create_user(username, email, password)
+        user.first_name, user.last_name = validated_data.pop('first_name'), validated_data.pop('last_name')
+        
+        user.birth_date = validated_data.pop('birth_date')
+        user.is_corporate_user = is_corporate_user
+
+        if is_corporate_user:
+            corp = CorporateUserProfile.objects.create(**corporate_profile)
+            user.corporate_profile = corp
+        else:
+            user.corporate_profile = None
+        return user
+
+
+class UserDetailsSerializer(serializers.ModelSerializer):
+    corporate_profile = CorporateUserSerializer()
+    interests = UserInterestSerializer(many=True)
+    medias = MediaDetailsSerializer(many=True)
+    comments = CommentDetailsSerializer(many=True, read_only=True)
+    votes = VoteDetailsSerializer(many=True, read_only=True)
+    followers_count = serializers.SerializerMethodField(read_only=True)
+    followings_count = serializers.SerializerMethodField(read_only=True)
+    blocked_users_count = serializers.SerializerMethodField(read_only=True)
+    owned_events_count = serializers.SerializerMethodField(read_only=True)
+
+    class Meta:
+        model = User
+        fields = ('username', 'first_name', 'last_name', 'birth_date',
+                  'interests', 'medias', 'comments', 'votes',
+                  'followers_count', 'followings_count', 'owned_events_count',
+                  'blocked_users_count', 'is_corporate_user', 'corporate_profile')
+        read_only_fields = ('username', 'interests', 'medias', 'comments', 'votes',
```

Comment:
Two things here.

1. Users can update their interests. Why is that read only?
2. You don't need to specify fields in `read_only_fields` if they're already specified so. (in the beginning of the class)